### ISSUE TITLE
Compatibility issue with cx_Freeze with Python 2.7

### ISSUE THREAD

**Issue by PierreRaybaut** [Julian/jsonschema#263]
Currently, `jsonschema` breaks any attempt to deploy a Python script using `cx_Freeze`, a common tool for deploying self-consistent Python applications, (not only, but for example) on Windows platforms.

The source of the problem is in [jsonschema/compat.py](https://github.com/Julian/jsonschema/blob/master/jsonschema/compat.py#L5):
```python
try:
    from collections import MutableMapping, Sequence  # noqa
except ImportError:
    from collections.abc import MutableMapping, Sequence  # noqa
```

When commenting those lines, `cx_Freeze` works as expected.

The fact is `jsonschema` is part of the huge dependency list of `IPython`. `IPython` is very invasive when installed: even if not needed by your application, deploying with `cx_Freeze` will take all `IPython` related modules with it (and `jsonschema` is among them).
In other words, building the most insignificant Python script will fail only because `IPython` is installed (which happens a lot) and it depends on `jsonschema`.

See PR #264

**Comment by Julian** [comment]
For anyone finding this: see the discussion on #264, which was a proposed fix (and a probably working one I assume :), for what seems like an upstream issue, but it'd need a test to be mergeable.

**Comment by Julian** [comment]
It seems like the upstream issue was fixed, so I'm going to close this as "unsupported, but willing to support if the above is addressed."
